File: C#/StringRepeat.cs
// Language: C#
// Level: 8kyu
// Name of Problem: String repeat

// Instructions: Write a function called repeatString which repeats the given String src exactly count times.

// Examples: 
// repeatStr(6, "I") // "IIIIII"
// repeatStr(5, "Hello") // "HelloHelloHelloHelloHello"

// Solution 1:
using System;
using System.Linq;

namespace Solution
{
  public static class Program
  {
    public static string repeatStr(int n, string s)
    {
      return String.Concat(Enumerable.Repeat(s, n));
    }
  }
}

// Sample Tests Passed:
// namespace Solution
// {
//   using NUnit.Framework;
//   using System;

//   [TestFixture]
//   public class SolutionTest
//   {
//     [Test]
//     public void MyTest()
//     {
//       Assert.AreEqual("***", Program.repeatStr(3, "*"));
//     }
//   }
// }

File: Javascript/StringRepeat.js
// Language: Javascript
// Level: 8kyu
// Name of Problem: String Repeat

// Instructions: Write a function called repeatString which repeats the given String src exactly count times.

// Examples:
repeatStr(6, "I") // "IIIIII"
repeatStr(5, "Hello") // "HelloHelloHelloHelloHello"

// Solution 1 using if/else statement:
function repeatStr(n, string) {
    if (n > 0) {
       return string.repeat(n)
     } else {
       return ""
     }
   }

// Solution 2 using ternary operator:
function repeatStr (n, str) {
    return n > 0 ? str.repeat(n) : ""
  }

// Sample Tests Passed:
describe("Tests", function() {
    it ("Basic tests", function() {
      Test.assertSimilar(repeatStr(3, "*"), "***");
      Test.assertSimilar(repeatStr(5, "#"), "#####");
      Test.assertSimilar(repeatStr(2, "ha "), "ha ha ");
    });
  });